homeassistant/components/roomba/binary_sensor.py

```diff
@@ -28,11 +28,18 @@ class RoombaBinStatus(BinarySensorDevice):
     def __init__(self, roomba, blid):
         """Initialize the sensor object."""
         self.vacuum = roomba
-        self.vacuum_state = roomba_reported_state(roomba)
+        vacuum_state = roomba_reported_state(roomba)
         self._blid = blid
-        self._name = self.vacuum_state.get("name")
+        self._name = vacuum_state.get("name")
         self._identifier = f"roomba_{self._blid}"
-        self._bin_status = None
+
+        # Register callback function
+        self.vacuum.register_on_message_callback(self.on_message)
```

Comment:
Please move side effects out of `__init__`. It can be another function that is called in `async_setup_entry` after the entity is created